docs/index.rst

```diff
@@ -129,54 +130,132 @@ Card Layout
 
-.. tip::
+.. seealso::
 
-    For card colouring, it is advised to use the bootstrap contextual classes:
-    `bg-primary`, `bg-success`, `bg-info`, `bg-warning`, `bg-danger`, `bg-secondary`, `bg-dark` and `bg-light`.
+    The bootstrap 4 `card documentation <https://getbootstrap.com/docs/4.0/components/card/>`_,
+    and this `card tutorial <https://www.w3schools.com/bootstrap4/bootstrap_cards.asp>`_
 
 .. code-block:: rst
 
     .. panels::
-        :card: shadow bg-primary
 
         panel 1 header
         ==============
 
         panel 1 content
 
-        ...
+        more content
+
+        ..............
         panel 1 footer
 
         ---
-        column += text-center
-        card = bg-info
-        title = bg-success
-        footer = bg-secondary
 
+        panel 2 header
+        ==============
+
+        panel 2 content
+
+        ..............
+        panel 2 footer
+
+.. panels::
+
+    panel 1 header
+    ==============
+
+    panel 1 content
+
+    more content
+
+    ..............
+    panel 1 footer
+
+    ---
+
+    panel 2 header
+    ==============
+
+    panel 2 content
+
+    ..............
+    panel 2 footer
+
+
+Card Styling
+------------
+
+To style the look of cards,
+you may use the directive options to add default CSS classes for eac element,
+or use the per-panel option syntax to add to or override these:
+
+- container: the top-level container
+- column: the panel container
+- card: the panel card
+- body: the panel card
+- header: the panel header
+- footer: the panel footer
+
+You can add your own CSS (see
+`the html_css_files option <https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-html_css_files>`_)
+but it is advised you use the built-in bootstrap classes:
+
+- `Card colouring <https://getbootstrap.com/docs/4.0/utilities/colors/>`_  contextual classes: `bg-primary`, `bg-success`, `bg-info`, `bg-warning`, `bg-danger`, `bg-secondary`, `bg-dark` and `bg-light`.
+- `Padding and margins <https://getbootstrap.com/docs/4.0/utilities/spacing/>`_: `border-0`, `p-2`, `m-2`, ...
+- `Text alignment <https://getbootstrap.com/docs/4.0/utilities/text/#text-alignment>`_: `text-justify`, `text-left`, `text-center`, `text-right`
+
+.. code-block:: rst
+
+    .. panels::
+        :body: bg-primary text-justify
+        :header: text-center
+        :footer: text-right
+
+        ---
+        column += p-1
+
+        panel 1 header
+        ==============
+
+        panel 1 content
+
+        ..............
+        panel 1 footer
+
+        ---
+        column += p-1 text-center border-0
+        body = bg-info
+        header = bg-success
+        footer = bg-secondary
 
         panel 2 header
         ==============
 
         panel 2 content
 
-        ...
+        ..............
         panel 2 footer
 
 .. panels::
-    :card: shadow bg-primary
+    :body: bg-primary text-justify
+    :header: text-center
+    :footer: text-right
+
+    ---
+    column += p-1
```

Comment:
It's unclear what's happening here - are you running your own python code inside of the panel content?

Reply:
just using regexes to identify <column><+=><p-1>, then deciding whether to add to the existing classes, or reset them based on if `=` or `+=`. See: https://github.com/executablebooks/sphinx-panels/blob/195e853a214df6234c107d5015b3213070bec42c/sphinx_panels/__init__.py#L134-L139